lib/holdingsiq.js

```diff
@@ -0,0 +1,345 @@
+      PublicationType: e?.publication_type,
+      ResourceType: e?.resource_type,
+      URL: e?.title_url,
+      Author: e?.first_author,
+      NumLastIssueOnline: e?.num_last_issue_online,
+      NumLastVolOnline: e?.num_last_vol_online,
+      CoverageDepth: e?.coverage_depth,
+      NumFirstIssueOnline: e?.num_first_issue_online,
+      NumFirstVolOnline: e?.num_first_vol_online,
+      DateFirstIssueOnline: e?.date_first_issue_online,
+      DateLastIssueOnline: e?.date_last_issue_online,
+      EmbargoInfo: e?.embargo_info,
+      OnlineIdentifier: e?.online_identifier,
+      PrintIdentifier: e?.print_identifier,
+    };
+    results.push({ update: { _index: index, _id: result.ezhlmid } });
+    results.push({ doc: result });
+  });
+  return results.slice();
+};
+
+/**
+ * getHoldings for enrich initialization
+ * @param {Object} conf - config on institute (name, apikey, custid)
+ * @param {*} count - number of documents to recover
+ * @param {*} offset - page
+ * @param {String} index Index where the values will be inserted
+ * @returns {Object} Data ready to be inserted in elastic
+ */
+const getHoldings = async (conf, count, offset, index) => {
+  const {
+    apikey,
+    custid,
+    name,
+  } = conf;
+
+  let res;
+
+  try {
+    res = await holdings({
+      method: 'get',
+      url: `/${custid}/holdings`,
+      params: { count, offset, format },
+      headers: {
+        'x-api-key': apikey,
+        'Content-Type': 'application/json',
+      },
+    });
+  } catch (err) {
+    logger.errorRequest(err);
+    process.exit(1);
+  }
+  const result = parseGetHoldings(res?.data?.holdings, name, index);
+  return result;
+};
+
+/**
+ * get the contributor according to the type
+ * @param {Array<Object>} contributors - contributors array
+ * @param {String} type - type of contributor
+ * @returns {String} name of contributor
+ */
+const getContributor = (contributors, type) => contributors.filter((contributor) => {
+  if (contributor?.type === type) {
+    return contributor?.value;
+  }
+});
+
+/**
+ * get the identifier according to the type and subtype
+ * @param {Array<Object>} identifiers - identifiers array
+ * @param {String} type - type of identifiers
+ * @param {String} subtype - subtype of identifiers
+ * @returns {String} name of contributor
+ */
+const getIdentifier = (identifiers, type, subtype) => identifiers.filter((identifier) => {
+  if (identifier?.type === type && identifier?.subtype === subtype) {
+    return identifier?.id;
+  }
+});
+
+/**
+ * parse embargo to ezHLM format
+ * @param {Object} embargo embargo
+ * @returns {String} embargo at ezhlm format
+ */
+const getEmbargo = (embargo) => {
+  if (embargo?.embargoValue === 0) {
+    return '0 mois';
+  }
+  let multiplicator = 1;
+  if (embargo?.embargoUnit === 'years') {
+    multiplicator = 12;
+  }
+  if (embargo?.embargoUnit === 'days') {
+    multiplicator = 0.3;
+  }
+  return `${embargo?.embargoUnit * multiplicator} mois`;
```

Comment:
You're multiplying a `string` with a `number` here. However, why not simply return "X years" or "X days" instead of converting everything into months ?

Another note : the correct multiplicator for the days is `0.03`. However `days / 30` would be easier to read. Also keep in mind that you can get something like `0.44999999999999996 mois` (`15 * 0.03`)

Another note : the function should return `NaN mois` if anything is not defined